logical/framework/openapi.go

```diff
@@ -0,0 +1,548 @@
+package framework
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/helper/wrapping"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/version"
+	"github.com/mitchellh/mapstructure"
+)
+
+// OpenAPI specification (OAS): https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md
+const OASVersion = "3.0.2"
+
+// NewOASDocument returns an empty OpenAPI document.
+func NewOASDocument() *OASDocument {
+	return &OASDocument{
+		Version: OASVersion,
+		Info: oasInfo{
+			Title:       "HashiCorp Vault API",
+			Description: "HTTP API that gives you full access to Vault. All API routes are prefixed with `/v1/`.",
+			Version:     version.GetVersion().Version,
+			License: oasLicense{
+				Name: "Mozilla Public License 2.0",
+				URL:  "https://www.mozilla.org/en-US/MPL/2.0",
+			},
+		},
+		Paths: make(map[string]*oasPathItem),
+	}
+}
+
+type OASDocument struct {
+	Version string                  `json:"openapi"`
+	Info    oasInfo                 `json:"info"`
+	Paths   map[string]*oasPathItem `json:"paths"`
+}
+
+type oasInfo struct {
+	Title       string     `json:"title"`
+	Description string     `json:"description"`
+	Version     string     `json:"version"`
+	License     oasLicense `json:"license"`
+}
+
+type oasLicense struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+}
+
+type oasPathItem struct {
+	Description     string         `json:"description,omitempty"`
+	Parameters      []oasParameter `json:"parameters,omitempty"`
+	Sudo            bool           `json:"x-vault-sudo,omitempty" mapstructure:"x-vault-sudo"`
+	Unauthenticated bool           `json:"x-vault-unauthenticated,omitempty" mapstructure:"x-vault-unauthenticated"`
+	Create          bool           `json:"x-vault-create,omitempty" mapstructure:"x-vault-create"`
```

Comment:
Maybe `CreateSupported`? Otherwise it feels like maybe people will be confused into thinking it's create _only_.

Reply:
I'll change it to `CreateSupported`.